pkg/osquery/tables/dataflattentable/exec_and_parse.go

```diff
@@ -0,0 +1,101 @@
+package dataflattentable
+
+import (
+	"context"
+	"os"
+	"strings"
+
+	"github.com/go-kit/kit/log"
+	"github.com/go-kit/kit/log/level"
+	"github.com/kolide/launcher/pkg/dataflatten"
+	"github.com/kolide/launcher/pkg/osquery/tables/tablehelpers"
+	"github.com/osquery/osquery-go/plugin/table"
+	"github.com/pkg/errors"
+)
+
+type flattenBytesInt interface {
+	FlattenBytes([]byte, ...dataflatten.FlattenOpts) ([]dataflatten.Row, error)
+}
+
+// execTableV2 is the next iteration of the dataflattentable wrapper. Aim to migrate exec based tables to this.
+type execTableV2 struct {
+	logger         log.Logger
+	tableName      string
+	flattener      flattenBytesInt
+	timeoutSeconds int
+	tabledebug     bool
+	execPaths      []string
+	execArgs       []string
+}
+
+type execTableV2Opt func(*execTableV2)
+
+func WithTimeout(ts int) execTableV2Opt {
+	return func(t *execTableV2) {
+		t.timeoutSeconds = ts
+	}
+}
+
+func WithTableDebug() execTableV2Opt {
+	return func(t *execTableV2) {
+		t.tabledebug = true
+	}
+}
+
+func WithAdditionalExecPaths(paths ...string) execTableV2Opt {
+	return func(t *execTableV2) {
+		t.execPaths = append(t.execPaths, paths...)
+	}
+}
+
+func NewExecAndParseTable(logger log.Logger, tableName string, parser parserInt, execCmd []string, opts ...execTableV2Opt) *table.Plugin {
+	t := &execTableV2{
+		logger:         level.NewFilter(log.With(logger, "table", tableName), level.AllowInfo()),
+		tableName:      tableName,
+		flattener:      flattenerFromParser(parser),
+		timeoutSeconds: 30,
+		execPaths:      execCmd[:1],
+		execArgs:       execCmd[1:],
```

Comment:
The extraction of the execPaths & execArgs feels weird to me .... I guess because I'm used to the standard lib command signature `func Command(name string, arg ...string) *Cmd`

Consider just accepting a `name` or `commandName` string param the adding a `func WithArgs(args ...string) execTableV2Opt` option?

Reply:
fair enough